sdk/core/Azure.Core/src/TelemetryDetails.cs

```diff
@@ -61,127 +55,37 @@ public void Apply(HttpMessage message)
             message.SetProperty(typeof(UserAgentValueKey), ToString());
         }
 
-        internal static string GenerateUserAgentString(Assembly clientAssembly, string? applicationId = null, RuntimeInformationWrapper? runtimeInformation = default)
+        internal static string GenerateUserAgentString(Assembly clientAssembly, string? applicationId = null)
         {
-            const string PackagePrefix = "Azure.";
+            // Use System.ClientModel's UserAgentPolicy.GenerateUserAgentString and apply Azure SDK specific transformations
+            string userAgentString = UserAgentPolicy.GenerateUserAgentString(clientAssembly, applicationId);
+            return ApplyAzureSdkTransformations(userAgentString, clientAssembly);
+        }
 
-            AssemblyInformationalVersionAttribute? versionAttribute = clientAssembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>();
-            if (versionAttribute == null)
-            {
-                throw new InvalidOperationException(
-                    $"{nameof(AssemblyInformationalVersionAttribute)} is required on client SDK assembly '{clientAssembly.FullName}'.");
-            }
+        private static string ApplyAzureSdkTransformations(string userAgentString, Assembly clientAssembly)
```

Comment:
```suggestion
        private static string ApplyAzureTransformations(string userAgentString, Assembly clientAssembly)
```

I think the SDK context is a given, just referencing `Azure` seems all that is needed to differentiate.